app/models/job_template.rb

```diff
@@ -37,14 +37,45 @@ def self.base_class
   end
   self.table_name = 'templates'
 
+  # Import a template ERB, with metadata in the first YAML comment
+  def self.import(template, options = {})
+    metadata = parse_metadata(template)
+    return if metadata.blank? || metadata.delete('kind') != 'job_template' || self.find_by_name(metadata['name'])
+
+    inputs = metadata.delete('template_inputs')
+
+    # This awkward dance is because you can't instantiate a new template with :locked => true
```

Comment:
would it make sense to fix it in core so locked attribute is checked only for persisted records? (nothing that would block this PR)